provider/sgx.py

```diff
@@ -0,0 +1,189 @@
+"""
+Module for sgx relevant operations.
+"""
+import logging
+import re
+
+from avocado.utils import process
+from virttest.staging import utils_memory
+from virttest.utils_misc import normalize_data_size
+
+LOG_JOB = logging.getLogger('avocado.test')
+
+
+class SGXError(Exception):
+    """ General SGX error"""
+    pass
+
+
+def _get_epc_size(output):
+    """
+    Get epc size from system message
+
+    :param output: epc section info
+    :return: total epc size
+    """
+    epc_size = 0
+    for line in output.splitlines():
+        tmp_epc = re.search(r'\b0x[0-9a-fA-F]*-0x[0-9a-fA-F]*\b',
+                            str(line))
+        if tmp_epc:
+            epc_size += int(tmp_epc[0].split('-')[1], 16) - int(
+                tmp_epc[0].split("-")[0], 16) + 1
+    return epc_size
+
+
+class SGXTestCapability(object):
+    """
+    Hypervisor sgx test capabilities check.
+    """
+
+    def __init__(self, test, params):
+        """
+
+        :param test: Context of test.
+        :param params: params of running ENV.
+        """
+        self._test = test
+        self._params = params
+
+    def valid_host_sgx_cap(self):
```

Comment:
so 'host' in the function name is not required, but it's OK for me to have it

What I consider here is a 'valid' function should not return something, i.e. it quits with errors when the function fails, so suggest to divide this function into two functions:
1. valid_sgx_cap
2. get_epc_size
in the 1st function, you can call process.system() and catch the exception, while in the 2nd function, you can use system_output function to get the output without any try/catch

Reply:
The purpose of this function is to validate if host has available epc, this will be used to verify qmp query_sgx_capabilities  and we can also use this host epc size to assign guest epc size dynamically in the other epc limit test case. I agree to split it into 2 functions, just in the first function valid_sgx_cap, I tend to save output msg to params, so when we need to get host epc size, we can call _get_epc_size(params.['host_sgx_msg']) without run system_ouput again.